{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Differential Expression Analysis with DEseq2\n",
    "We have quantified the transcript expression using `Salmon` which "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: S4Vectors\n",
      "\n",
      "Loading required package: stats4\n",
      "\n",
      "Loading required package: BiocGenerics\n",
      "\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which.max, which.min\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "Loading required package: GenomicRanges\n",
      "\n",
      "Loading required package: GenomeInfoDb\n",
      "\n",
      "Loading required package: SummarizedExperiment\n",
      "\n",
      "Loading required package: MatrixGenerics\n",
      "\n",
      "Loading required package: matrixStats\n",
      "\n",
      "\n",
      "Attaching package: ‘MatrixGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,\n",
      "    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,\n",
      "    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,\n",
      "    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,\n",
      "    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,\n",
      "    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,\n",
      "    colWeightedMeans, colWeightedMedians, colWeightedSds,\n",
      "    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,\n",
      "    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,\n",
      "    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,\n",
      "    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,\n",
      "    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,\n",
      "    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,\n",
      "    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,\n",
      "    rowWeightedSds, rowWeightedVars\n",
      "\n",
      "\n",
      "Loading required package: Biobase\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘Biobase’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:MatrixGenerics’:\n",
      "\n",
      "    rowMedians\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    anyMissing, rowMedians\n",
      "\n",
      "\n",
      "Loading required package: limma\n",
      "\n",
      "\n",
      "Attaching package: ‘limma’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:DESeq2’:\n",
      "\n",
      "    plotMA\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:BiocGenerics’:\n",
      "\n",
      "    plotMA\n",
      "\n",
      "\n",
      "Loading required package: AnnotationDbi\n",
      "\n",
      "Import genomic features from the file as a GRanges object ... \n",
      "OK\n",
      "\n",
      "Prepare the 'metadata' data frame ... \n",
      "OK\n",
      "\n",
      "Make the TxDb object ... \n",
      "OK\n",
      "\n",
      "'select()' returned 1:1 mapping between keys and columns\n",
      "\n"
     ]
    }
   ],
   "source": [
    "\n",
    "#.libPaths('/home/ualtintas/miniconda3/envs/notebook/lib/R/library')\n",
    "options(warn=-1)\n",
    "options(repr.matrix.max.cols=100, repr.matrix.max.rows=10)\n",
    "\n",
    "library(DESeq2)\n",
    "library(edgeR)\n",
    "library(tximport)\n",
    "library(apeglm)\n",
    "library(GenomicFeatures)\n",
    "\n",
    "\n",
    "txdb <- makeTxDbFromGFF(file=\"/groups/lackgrp/genomeAnnotations/hg38/hg38.ncbiRefSeq.simple.gtf\")\n",
    "k <- keys(txdb, keytype = \"TXNAME\")\n",
    "tx2gene <- select(txdb, k, \"GENEID\", \"TXNAME\") "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "fns = list.files(\"rnaseq-pipeline/results_hg38/salmon\", pattern = \"quant.sf\", recursive = T, full.names = T)\n",
    "\n",
    "names(fns) = basename(dirname(fns))\n",
    "\n",
    "metaDataFile = \"data/rna/metadata.tsv\"\n",
    "metaData = read.table(metaDataFile,header=TRUE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>AR_rep1</dt><dd>'rnaseq-pipeline/results_hg38/salmon/AR_rep1/quant.sf'</dd><dt>AR_rep2</dt><dd>'rnaseq-pipeline/results_hg38/salmon/AR_rep2/quant.sf'</dd><dt>AR_rep3</dt><dd>'rnaseq-pipeline/results_hg38/salmon/AR_rep3/quant.sf'</dd><dt>GSM3502415_MR49F_1</dt><dd>'rnaseq-pipeline/results_hg38/salmon/GSM3502415_MR49F_1/quant.sf'</dd><dt>GSM3502416_MR49F_2</dt><dd>'rnaseq-pipeline/results_hg38/salmon/GSM3502416_MR49F_2/quant.sf'</dd><dt>GSM3502417_MR49F_3</dt><dd>'rnaseq-pipeline/results_hg38/salmon/GSM3502417_MR49F_3/quant.sf'</dd><dt>GSM3502418_LNCaP_1</dt><dd>'rnaseq-pipeline/results_hg38/salmon/GSM3502418_LNCaP_1/quant.sf'</dd><dt>GSM3502419_LNCaP_2</dt><dd>'rnaseq-pipeline/results_hg38/salmon/GSM3502419_LNCaP_2/quant.sf'</dd><dt>GSM3502420_LNCaP_3</dt><dd>'rnaseq-pipeline/results_hg38/salmon/GSM3502420_LNCaP_3/quant.sf'</dd><dt>GSM4777223_LNCaP_1</dt><dd>'rnaseq-pipeline/results_hg38/salmon/GSM4777223_LNCaP_1/quant.sf'</dd><dt>GSM4777224_LNCaP_2</dt><dd>'rnaseq-pipeline/results_hg38/salmon/GSM4777224_LNCaP_2/quant.sf'</dd><dt>GSM4777231_MR49F_1</dt><dd>'rnaseq-pipeline/results_hg38/salmon/GSM4777231_MR49F_1/quant.sf'</dd><dt>GSM4777232_MR49F_2</dt><dd>'rnaseq-pipeline/results_hg38/salmon/GSM4777232_MR49F_2/quant.sf'</dd><dt>NT_rep1</dt><dd>'rnaseq-pipeline/results_hg38/salmon/NT_rep1/quant.sf'</dd><dt>NT_rep2</dt><dd>'rnaseq-pipeline/results_hg38/salmon/NT_rep2/quant.sf'</dd><dt>NT_rep3</dt><dd>'rnaseq-pipeline/results_hg38/salmon/NT_rep3/quant.sf'</dd><dt>SMARCC2sg1_rep1</dt><dd>'rnaseq-pipeline/results_hg38/salmon/SMARCC2sg1_rep1/quant.sf'</dd><dt>SMARCC2sg1_rep2</dt><dd>'rnaseq-pipeline/results_hg38/salmon/SMARCC2sg1_rep2/quant.sf'</dd><dt>SMARCC2sg1_rep3</dt><dd>'rnaseq-pipeline/results_hg38/salmon/SMARCC2sg1_rep3/quant.sf'</dd><dt>SMARCC2sg2_rep1</dt><dd>'rnaseq-pipeline/results_hg38/salmon/SMARCC2sg2_rep1/quant.sf'</dd><dt>SMARCC2sg2_rep2</dt><dd>'rnaseq-pipeline/results_hg38/salmon/SMARCC2sg2_rep2/quant.sf'</dd><dt>SMARCC2sg2_rep3</dt><dd>'rnaseq-pipeline/results_hg38/salmon/SMARCC2sg2_rep3/quant.sf'</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[AR\\textbackslash{}\\_rep1] 'rnaseq-pipeline/results\\_hg38/salmon/AR\\_rep1/quant.sf'\n",
       "\\item[AR\\textbackslash{}\\_rep2] 'rnaseq-pipeline/results\\_hg38/salmon/AR\\_rep2/quant.sf'\n",
       "\\item[AR\\textbackslash{}\\_rep3] 'rnaseq-pipeline/results\\_hg38/salmon/AR\\_rep3/quant.sf'\n",
       "\\item[GSM3502415\\textbackslash{}\\_MR49F\\textbackslash{}\\_1] 'rnaseq-pipeline/results\\_hg38/salmon/GSM3502415\\_MR49F\\_1/quant.sf'\n",
       "\\item[GSM3502416\\textbackslash{}\\_MR49F\\textbackslash{}\\_2] 'rnaseq-pipeline/results\\_hg38/salmon/GSM3502416\\_MR49F\\_2/quant.sf'\n",
       "\\item[GSM3502417\\textbackslash{}\\_MR49F\\textbackslash{}\\_3] 'rnaseq-pipeline/results\\_hg38/salmon/GSM3502417\\_MR49F\\_3/quant.sf'\n",
       "\\item[GSM3502418\\textbackslash{}\\_LNCaP\\textbackslash{}\\_1] 'rnaseq-pipeline/results\\_hg38/salmon/GSM3502418\\_LNCaP\\_1/quant.sf'\n",
       "\\item[GSM3502419\\textbackslash{}\\_LNCaP\\textbackslash{}\\_2] 'rnaseq-pipeline/results\\_hg38/salmon/GSM3502419\\_LNCaP\\_2/quant.sf'\n",
       "\\item[GSM3502420\\textbackslash{}\\_LNCaP\\textbackslash{}\\_3] 'rnaseq-pipeline/results\\_hg38/salmon/GSM3502420\\_LNCaP\\_3/quant.sf'\n",
       "\\item[GSM4777223\\textbackslash{}\\_LNCaP\\textbackslash{}\\_1] 'rnaseq-pipeline/results\\_hg38/salmon/GSM4777223\\_LNCaP\\_1/quant.sf'\n",
       "\\item[GSM4777224\\textbackslash{}\\_LNCaP\\textbackslash{}\\_2] 'rnaseq-pipeline/results\\_hg38/salmon/GSM4777224\\_LNCaP\\_2/quant.sf'\n",
       "\\item[GSM4777231\\textbackslash{}\\_MR49F\\textbackslash{}\\_1] 'rnaseq-pipeline/results\\_hg38/salmon/GSM4777231\\_MR49F\\_1/quant.sf'\n",
       "\\item[GSM4777232\\textbackslash{}\\_MR49F\\textbackslash{}\\_2] 'rnaseq-pipeline/results\\_hg38/salmon/GSM4777232\\_MR49F\\_2/quant.sf'\n",
       "\\item[NT\\textbackslash{}\\_rep1] 'rnaseq-pipeline/results\\_hg38/salmon/NT\\_rep1/quant.sf'\n",
       "\\item[NT\\textbackslash{}\\_rep2] 'rnaseq-pipeline/results\\_hg38/salmon/NT\\_rep2/quant.sf'\n",
       "\\item[NT\\textbackslash{}\\_rep3] 'rnaseq-pipeline/results\\_hg38/salmon/NT\\_rep3/quant.sf'\n",
       "\\item[SMARCC2sg1\\textbackslash{}\\_rep1] 'rnaseq-pipeline/results\\_hg38/salmon/SMARCC2sg1\\_rep1/quant.sf'\n",
       "\\item[SMARCC2sg1\\textbackslash{}\\_rep2] 'rnaseq-pipeline/results\\_hg38/salmon/SMARCC2sg1\\_rep2/quant.sf'\n",
       "\\item[SMARCC2sg1\\textbackslash{}\\_rep3] 'rnaseq-pipeline/results\\_hg38/salmon/SMARCC2sg1\\_rep3/quant.sf'\n",
       "\\item[SMARCC2sg2\\textbackslash{}\\_rep1] 'rnaseq-pipeline/results\\_hg38/salmon/SMARCC2sg2\\_rep1/quant.sf'\n",
       "\\item[SMARCC2sg2\\textbackslash{}\\_rep2] 'rnaseq-pipeline/results\\_hg38/salmon/SMARCC2sg2\\_rep2/quant.sf'\n",
       "\\item[SMARCC2sg2\\textbackslash{}\\_rep3] 'rnaseq-pipeline/results\\_hg38/salmon/SMARCC2sg2\\_rep3/quant.sf'\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "AR_rep1\n",
       ":   'rnaseq-pipeline/results_hg38/salmon/AR_rep1/quant.sf'AR_rep2\n",
       ":   'rnaseq-pipeline/results_hg38/salmon/AR_rep2/quant.sf'AR_rep3\n",
       ":   'rnaseq-pipeline/results_hg38/salmon/AR_rep3/quant.sf'GSM3502415_MR49F_1\n",
       ":   'rnaseq-pipeline/results_hg38/salmon/GSM3502415_MR49F_1/quant.sf'GSM3502416_MR49F_2\n",
       ":   'rnaseq-pipeline/results_hg38/salmon/GSM3502416_MR49F_2/quant.sf'GSM3502417_MR49F_3\n",
       ":   'rnaseq-pipeline/results_hg38/salmon/GSM3502417_MR49F_3/quant.sf'GSM3502418_LNCaP_1\n",
       ":   'rnaseq-pipeline/results_hg38/salmon/GSM3502418_LNCaP_1/quant.sf'GSM3502419_LNCaP_2\n",
       ":   'rnaseq-pipeline/results_hg38/salmon/GSM3502419_LNCaP_2/quant.sf'GSM3502420_LNCaP_3\n",
       ":   'rnaseq-pipeline/results_hg38/salmon/GSM3502420_LNCaP_3/quant.sf'GSM4777223_LNCaP_1\n",
       ":   'rnaseq-pipeline/results_hg38/salmon/GSM4777223_LNCaP_1/quant.sf'GSM4777224_LNCaP_2\n",
       ":   'rnaseq-pipeline/results_hg38/salmon/GSM4777224_LNCaP_2/quant.sf'GSM4777231_MR49F_1\n",
       ":   'rnaseq-pipeline/results_hg38/salmon/GSM4777231_MR49F_1/quant.sf'GSM4777232_MR49F_2\n",
       ":   'rnaseq-pipeline/results_hg38/salmon/GSM4777232_MR49F_2/quant.sf'NT_rep1\n",
       ":   'rnaseq-pipeline/results_hg38/salmon/NT_rep1/quant.sf'NT_rep2\n",
       ":   'rnaseq-pipeline/results_hg38/salmon/NT_rep2/quant.sf'NT_rep3\n",
       ":   'rnaseq-pipeline/results_hg38/salmon/NT_rep3/quant.sf'SMARCC2sg1_rep1\n",
       ":   'rnaseq-pipeline/results_hg38/salmon/SMARCC2sg1_rep1/quant.sf'SMARCC2sg1_rep2\n",
       ":   'rnaseq-pipeline/results_hg38/salmon/SMARCC2sg1_rep2/quant.sf'SMARCC2sg1_rep3\n",
       ":   'rnaseq-pipeline/results_hg38/salmon/SMARCC2sg1_rep3/quant.sf'SMARCC2sg2_rep1\n",
       ":   'rnaseq-pipeline/results_hg38/salmon/SMARCC2sg2_rep1/quant.sf'SMARCC2sg2_rep2\n",
       ":   'rnaseq-pipeline/results_hg38/salmon/SMARCC2sg2_rep2/quant.sf'SMARCC2sg2_rep3\n",
       ":   'rnaseq-pipeline/results_hg38/salmon/SMARCC2sg2_rep3/quant.sf'\n",
       "\n"
      ],
      "text/plain": [
       "                                                          AR_rep1 \n",
       "           \"rnaseq-pipeline/results_hg38/salmon/AR_rep1/quant.sf\" \n",
       "                                                          AR_rep2 \n",
       "           \"rnaseq-pipeline/results_hg38/salmon/AR_rep2/quant.sf\" \n",
       "                                                          AR_rep3 \n",
       "           \"rnaseq-pipeline/results_hg38/salmon/AR_rep3/quant.sf\" \n",
       "                                               GSM3502415_MR49F_1 \n",
       "\"rnaseq-pipeline/results_hg38/salmon/GSM3502415_MR49F_1/quant.sf\" \n",
       "                                               GSM3502416_MR49F_2 \n",
       "\"rnaseq-pipeline/results_hg38/salmon/GSM3502416_MR49F_2/quant.sf\" \n",
       "                                               GSM3502417_MR49F_3 \n",
       "\"rnaseq-pipeline/results_hg38/salmon/GSM3502417_MR49F_3/quant.sf\" \n",
       "                                               GSM3502418_LNCaP_1 \n",
       "\"rnaseq-pipeline/results_hg38/salmon/GSM3502418_LNCaP_1/quant.sf\" \n",
       "                                               GSM3502419_LNCaP_2 \n",
       "\"rnaseq-pipeline/results_hg38/salmon/GSM3502419_LNCaP_2/quant.sf\" \n",
       "                                               GSM3502420_LNCaP_3 \n",
       "\"rnaseq-pipeline/results_hg38/salmon/GSM3502420_LNCaP_3/quant.sf\" \n",
       "                                               GSM4777223_LNCaP_1 \n",
       "\"rnaseq-pipeline/results_hg38/salmon/GSM4777223_LNCaP_1/quant.sf\" \n",
       "                                               GSM4777224_LNCaP_2 \n",
       "\"rnaseq-pipeline/results_hg38/salmon/GSM4777224_LNCaP_2/quant.sf\" \n",
       "                                               GSM4777231_MR49F_1 \n",
       "\"rnaseq-pipeline/results_hg38/salmon/GSM4777231_MR49F_1/quant.sf\" \n",
       "                                               GSM4777232_MR49F_2 \n",
       "\"rnaseq-pipeline/results_hg38/salmon/GSM4777232_MR49F_2/quant.sf\" \n",
       "                                                          NT_rep1 \n",
       "           \"rnaseq-pipeline/results_hg38/salmon/NT_rep1/quant.sf\" \n",
       "                                                          NT_rep2 \n",
       "           \"rnaseq-pipeline/results_hg38/salmon/NT_rep2/quant.sf\" \n",
       "                                                          NT_rep3 \n",
       "           \"rnaseq-pipeline/results_hg38/salmon/NT_rep3/quant.sf\" \n",
       "                                                  SMARCC2sg1_rep1 \n",
       "   \"rnaseq-pipeline/results_hg38/salmon/SMARCC2sg1_rep1/quant.sf\" \n",
       "                                                  SMARCC2sg1_rep2 \n",
       "   \"rnaseq-pipeline/results_hg38/salmon/SMARCC2sg1_rep2/quant.sf\" \n",
       "                                                  SMARCC2sg1_rep3 \n",
       "   \"rnaseq-pipeline/results_hg38/salmon/SMARCC2sg1_rep3/quant.sf\" \n",
       "                                                  SMARCC2sg2_rep1 \n",
       "   \"rnaseq-pipeline/results_hg38/salmon/SMARCC2sg2_rep1/quant.sf\" \n",
       "                                                  SMARCC2sg2_rep2 \n",
       "   \"rnaseq-pipeline/results_hg38/salmon/SMARCC2sg2_rep2/quant.sf\" \n",
       "                                                  SMARCC2sg2_rep3 \n",
       "   \"rnaseq-pipeline/results_hg38/salmon/SMARCC2sg2_rep3/quant.sf\" "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "fns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "reading in files with read_tsv\n",
      "\n",
      "1 \n",
      "2 \n",
      "3 \n",
      "4 \n",
      "5 \n",
      "6 \n",
      "\n",
      "\n",
      "summarizing abundance\n",
      "\n",
      "summarizing counts\n",
      "\n",
      "summarizing length\n",
      "\n",
      "using counts and average transcript lengths from tximport\n",
      "\n",
      "estimating size factors\n",
      "\n",
      "using 'avgTxLength' from assays(dds), correcting for library size\n",
      "\n",
      "estimating dispersions\n",
      "\n",
      "gene-wise dispersion estimates\n",
      "\n",
      "mean-dispersion relationship\n",
      "\n",
      "-- note: fitType='parametric', but the dispersion trend was not well captured by the\n",
      "   function: y = a/x + b, and a local regression fit was automatically substituted.\n",
      "   specify fitType='local' or 'mean' to avoid this message next time.\n",
      "\n",
      "final dispersion estimates\n",
      "\n",
      "fitting model and testing\n",
      "\n",
      "3 rows did not converge in beta, labelled in mcols(object)$betaConv. Use larger maxit argument with nbinomWaldTest\n",
      "\n",
      "using 'apeglm' for LFC shrinkage. If used in published research, please cite:\n",
      "    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for\n",
      "    sequence count data: removing the noise and preserving large differences.\n",
      "    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "out of 19016 with nonzero total read count\n",
      "adjusted p-value < 0.1\n",
      "LFC > 0 (up)       : 2482, 13%\n",
      "LFC < 0 (down)     : 2421, 13%\n",
      "outliers [1]       : 5, 0.026%\n",
      "low counts [2]     : 2212, 12%\n",
      "(mean count < 4)\n",
      "[1] see 'cooksCutoff' argument of ?results\n",
      "[2] see 'independentFiltering' argument of ?results\n",
      "\n"
     ]
    }
   ],
   "source": [
    "\n",
    "idx = (metaData$Target %in% c('NT', 'AR')) & (metaData$Dataset == 'BG')\n",
    "\n",
    "txi = tximport(fns[idx], type = \"salmon\",tx2gene = tx2gene)\n",
    "\n",
    "dds <- DESeqDataSetFromTximport(txi, metaData[idx,], ~Target)\n",
    "\n",
    "keep <- rowSums(counts(dds)) >= 10\n",
    "dds <- dds[keep,]\n",
    "\n",
    "dds$Target = relevel(dds$Target, ref = \"NT\")\n",
    "\n",
    "dds$Target <- droplevels(dds$Target)\n",
    "\n",
    "dds <- DESeq(dds)\n",
    "\n",
    "resLFC = lfcShrink(dds, coef=\"Target_AR_vs_NT\",type=\"apeglm\")\n",
    "summary(resLFC)\n",
    "\n",
    "write.table(file=\"data/rna/deg/AR.tsv\", x=as.data.frame(resLFC), sep=\"\\t\", quote = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "reading in files with read_tsv\n",
      "\n",
      "1 \n",
      "2 \n",
      "3 \n",
      "4 \n",
      "5 \n",
      "6 \n",
      "\n",
      "\n",
      "summarizing abundance\n",
      "\n",
      "summarizing counts\n",
      "\n",
      "summarizing length\n",
      "\n",
      "using counts and average transcript lengths from tximport\n",
      "\n",
      "estimating size factors\n",
      "\n",
      "using 'avgTxLength' from assays(dds), correcting for library size\n",
      "\n",
      "estimating dispersions\n",
      "\n",
      "gene-wise dispersion estimates\n",
      "\n",
      "mean-dispersion relationship\n",
      "\n",
      "-- note: fitType='parametric', but the dispersion trend was not well captured by the\n",
      "   function: y = a/x + b, and a local regression fit was automatically substituted.\n",
      "   specify fitType='local' or 'mean' to avoid this message next time.\n",
      "\n",
      "final dispersion estimates\n",
      "\n",
      "fitting model and testing\n",
      "\n",
      "4 rows did not converge in beta, labelled in mcols(object)$betaConv. Use larger maxit argument with nbinomWaldTest\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# TODO : metadata\n",
    "idx = (metaData$Target %in% c('NT', 'SMARCC2sg1'))\n",
    "\n",
    "txi = tximport(fns[idx], type = \"salmon\",tx2gene = tx2gene)\n",
    "\n",
    "dds <- DESeqDataSetFromTximport(txi, metaData[idx,], ~Target)\n",
    "\n",
    "keep <- rowSums(counts(dds)) >= 10\n",
    "dds <- dds[keep,]\n",
    "\n",
    "dds$Target = relevel(dds$Target, ref = \"NT\")\n",
    "\n",
    "dds$Target <- droplevels(dds$Target)\n",
    "\n",
    "dds <- DESeq(dds)\n",
    "\n",
    "\n",
    "resLFC = lfcShrink(dds, coef=\"Target_SMARCC2sg1_vs_NT\",type=\"apeglm\")\n",
    "summary(resLFC)\n",
    "\n",
    "\n",
    "write.table(file=\"data/rna/deg/SMARCC2sg1.tsv\", x=as.data.frame(resLFC), sep=\"\\t\", quote = FALSE)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "using 'apeglm' for LFC shrinkage. If used in published research, please cite:\n",
      "    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for\n",
      "    sequence count data: removing the noise and preserving large differences.\n",
      "    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "out of 18926 with nonzero total read count\n",
      "adjusted p-value < 0.1\n",
      "LFC > 0 (up)       : 671, 3.5%\n",
      "LFC < 0 (down)     : 559, 3%\n",
      "outliers [1]       : 8, 0.042%\n",
      "low counts [2]     : 3670, 19%\n",
      "(mean count < 8)\n",
      "[1] see 'cooksCutoff' argument of ?results\n",
      "[2] see 'independentFiltering' argument of ?results\n",
      "\n"
     ]
    }
   ],
   "source": [
    "\n",
    "resLFC = lfcShrink(dds, coef=\"Target_SMARCC2sg1_vs_NT\",type=\"apeglm\")\n",
    "summary(resLFC)\n",
    "\n",
    "\n",
    "write.table(file=\"data/rna/deg/SMARCC2sg1.tsv\", x=as.data.frame(resLFC), sep=\"\\t\", quote = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "reading in files with read_tsv\n",
      "\n",
      "1 \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2 \n",
      "3 \n",
      "4 \n",
      "5 \n",
      "6 \n",
      "\n",
      "\n",
      "summarizing abundance\n",
      "\n",
      "summarizing counts\n",
      "\n",
      "summarizing length\n",
      "\n",
      "using counts and average transcript lengths from tximport\n",
      "\n",
      "estimating size factors\n",
      "\n",
      "using 'avgTxLength' from assays(dds), correcting for library size\n",
      "\n",
      "estimating dispersions\n",
      "\n",
      "gene-wise dispersion estimates\n",
      "\n",
      "mean-dispersion relationship\n",
      "\n",
      "-- note: fitType='parametric', but the dispersion trend was not well captured by the\n",
      "   function: y = a/x + b, and a local regression fit was automatically substituted.\n",
      "   specify fitType='local' or 'mean' to avoid this message next time.\n",
      "\n",
      "final dispersion estimates\n",
      "\n",
      "fitting model and testing\n",
      "\n",
      "5 rows did not converge in beta, labelled in mcols(object)$betaConv. Use larger maxit argument with nbinomWaldTest\n",
      "\n",
      "using 'apeglm' for LFC shrinkage. If used in published research, please cite:\n",
      "    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for\n",
      "    sequence count data: removing the noise and preserving large differences.\n",
      "    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "out of 18944 with nonzero total read count\n",
      "adjusted p-value < 0.1\n",
      "LFC > 0 (up)       : 595, 3.1%\n",
      "LFC < 0 (down)     : 562, 3%\n",
      "outliers [1]       : 6, 0.032%\n",
      "low counts [2]     : 4407, 23%\n",
      "(mean count < 12)\n",
      "[1] see 'cooksCutoff' argument of ?results\n",
      "[2] see 'independentFiltering' argument of ?results\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# TODO : metadata\n",
    "idx = (metaData$Target %in% c('NT', 'SMARCC2sg2'))\n",
    "\n",
    "txi = tximport(fns[idx], type = \"salmon\",tx2gene = tx2gene)\n",
    "\n",
    "dds <- DESeqDataSetFromTximport(txi, metaData[idx,], ~Target)\n",
    "\n",
    "keep <- rowSums(counts(dds)) >= 10\n",
    "dds <- dds[keep,]\n",
    "\n",
    "dds$Target = relevel(dds$Target, ref = \"NT\")\n",
    "\n",
    "dds$Target <- droplevels(dds$Target)\n",
    "\n",
    "dds <- DESeq(dds)\n",
    "\n",
    "resLFC = lfcShrink(dds, coef=\"Target_SMARCC2sg2_vs_NT\",type=\"apeglm\")\n",
    "summary(resLFC)\n",
    "\n",
    "\n",
    "write.table(file=\"data/rna/deg/SMARCC2sg2.tsv\", x=as.data.frame(resLFC), sep=\"\\t\", quote = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "reading in files with read_tsv\n",
      "\n",
      "1 \n",
      "2 \n",
      "3 \n",
      "4 \n",
      "5 \n",
      "6 \n",
      "\n",
      "\n",
      "summarizing abundance\n",
      "\n",
      "summarizing counts\n",
      "\n",
      "summarizing length\n",
      "\n",
      "using counts and average transcript lengths from tximport\n",
      "\n",
      "estimating size factors\n",
      "\n",
      "using 'avgTxLength' from assays(dds), correcting for library size\n",
      "\n",
      "estimating dispersions\n",
      "\n",
      "gene-wise dispersion estimates\n",
      "\n",
      "mean-dispersion relationship\n",
      "\n",
      "-- note: fitType='parametric', but the dispersion trend was not well captured by the\n",
      "   function: y = a/x + b, and a local regression fit was automatically substituted.\n",
      "   specify fitType='local' or 'mean' to avoid this message next time.\n",
      "\n",
      "final dispersion estimates\n",
      "\n",
      "fitting model and testing\n",
      "\n",
      "8 rows did not converge in beta, labelled in mcols(object)$betaConv. Use larger maxit argument with nbinomWaldTest\n",
      "\n",
      "using 'apeglm' for LFC shrinkage. If used in published research, please cite:\n",
      "    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for\n",
      "    sequence count data: removing the noise and preserving large differences.\n",
      "    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "out of 16622 with nonzero total read count\n",
      "adjusted p-value < 0.1\n",
      "LFC > 0 (up)       : 4148, 25%\n",
      "LFC < 0 (down)     : 4043, 24%\n",
      "outliers [1]       : 4, 0.024%\n",
      "low counts [2]     : 0, 0%\n",
      "(mean count < 1)\n",
      "[1] see 'cooksCutoff' argument of ?results\n",
      "[2] see 'independentFiltering' argument of ?results\n",
      "\n"
     ]
    }
   ],
   "source": [
    "\n",
    "idx = (metaData$Target %in% c('LNCaP', 'MR49F')) & (metaData$Dataset == 'GSE123379')\n",
    "\n",
    "txi = tximport(fns[idx], type = \"salmon\",tx2gene = tx2gene)\n",
    "\n",
    "dds <- DESeqDataSetFromTximport(txi, metaData[idx,], ~Target)\n",
    "\n",
    "keep <- rowSums(counts(dds)) >= 10\n",
    "dds <- dds[keep,]\n",
    "\n",
    "dds$Target = relevel(dds$Target, ref = \"LNCaP\")\n",
    "\n",
    "dds$Target <- droplevels(dds$Target)\n",
    "\n",
    "dds <- DESeq(dds)\n",
    "\n",
    "resLFC = lfcShrink(dds, coef=\"Target_MR49F_vs_LNCaP\",type=\"apeglm\")\n",
    "summary(resLFC)\n",
    "\n",
    "\n",
    "write.table(file=\"data/rna/deg/GSE123379.tsv\", x=as.data.frame(resLFC), sep=\"\\t\", quote = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "reading in files with read_tsv\n",
      "\n",
      "1 \n",
      "2 \n",
      "3 \n",
      "4 \n",
      "\n",
      "\n",
      "summarizing abundance\n",
      "\n",
      "summarizing counts\n",
      "\n",
      "summarizing length\n",
      "\n",
      "using counts and average transcript lengths from tximport\n",
      "\n",
      "estimating size factors\n",
      "\n",
      "using 'avgTxLength' from assays(dds), correcting for library size\n",
      "\n",
      "estimating dispersions\n",
      "\n",
      "gene-wise dispersion estimates\n",
      "\n",
      "mean-dispersion relationship\n",
      "\n",
      "-- note: fitType='parametric', but the dispersion trend was not well captured by the\n",
      "   function: y = a/x + b, and a local regression fit was automatically substituted.\n",
      "   specify fitType='local' or 'mean' to avoid this message next time.\n",
      "\n",
      "final dispersion estimates\n",
      "\n",
      "fitting model and testing\n",
      "\n",
      "7 rows did not converge in beta, labelled in mcols(object)$betaConv. Use larger maxit argument with nbinomWaldTest\n",
      "\n",
      "using 'apeglm' for LFC shrinkage. If used in published research, please cite:\n",
      "    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for\n",
      "    sequence count data: removing the noise and preserving large differences.\n",
      "    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "out of 17659 with nonzero total read count\n",
      "adjusted p-value < 0.1\n",
      "LFC > 0 (up)       : 2052, 12%\n",
      "LFC < 0 (down)     : 1970, 11%\n",
      "outliers [1]       : 0, 0%\n",
      "low counts [2]     : 343, 1.9%\n",
      "(mean count < 3)\n",
      "[1] see 'cooksCutoff' argument of ?results\n",
      "[2] see 'independentFiltering' argument of ?results\n",
      "\n"
     ]
    }
   ],
   "source": [
    "\n",
    "idx = (metaData$Target %in% c('LNCaP', 'MR49F')) & (metaData$Dataset == 'GSE138460')\n",
    "\n",
    "txi = tximport(fns[idx], type = \"salmon\",tx2gene = tx2gene)\n",
    "\n",
    "dds <- DESeqDataSetFromTximport(txi, metaData[idx,], ~Target)\n",
    "\n",
    "keep <- rowSums(counts(dds)) >= 10\n",
    "dds <- dds[keep,]\n",
    "\n",
    "dds$Target = relevel(dds$Target, ref = \"LNCaP\")\n",
    "\n",
    "dds$Target <- droplevels(dds$Target)\n",
    "\n",
    "dds <- DESeq(dds)\n",
    "\n",
    "resLFC = lfcShrink(dds, coef=\"Target_MR49F_vs_LNCaP\",type=\"apeglm\")\n",
    "summary(resLFC)\n",
    "\n",
    "\n",
    "write.table(file=\"data/rna/deg/GSE138460.tsv\", x=as.data.frame(resLFC), sep=\"\\t\", quote = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "reading in files with read_tsv\n",
      "\n",
      "1 \n",
      "2 \n",
      "3 \n",
      "4 \n",
      "5 \n",
      "6 \n",
      "7 \n",
      "8 \n",
      "9 \n",
      "10 \n",
      "11 \n",
      "12 \n",
      "\n",
      "\n",
      "summarizing abundance\n",
      "\n",
      "summarizing counts\n",
      "\n",
      "summarizing length\n",
      "\n",
      "using counts and average transcript lengths from tximport\n",
      "\n",
      "estimating size factors\n",
      "\n",
      "using 'avgTxLength' from assays(dds), correcting for library size\n",
      "\n",
      "estimating dispersions\n",
      "\n",
      "gene-wise dispersion estimates\n",
      "\n",
      "mean-dispersion relationship\n",
      "\n",
      "-- note: fitType='parametric', but the dispersion trend was not well captured by the\n",
      "   function: y = a/x + b, and a local regression fit was automatically substituted.\n",
      "   specify fitType='local' or 'mean' to avoid this message next time.\n",
      "\n",
      "final dispersion estimates\n",
      "\n",
      "fitting model and testing\n",
      "\n"
     ]
    }
   ],
   "source": [
    "\n",
    "idx = (metaData$Target %in% c('NT', 'AR', 'SMARCC2sg1', 'SMARCC2sg2'))\n",
    "\n",
    "txi = tximport(fns[idx], type = \"salmon\",tx2gene = tx2gene)\n",
    "\n",
    "dds <- DESeqDataSetFromTximport(txi, metaData[idx,], ~Target)\n",
    "\n",
    "keep <- rowSums(counts(dds)) >= 10\n",
    "dds <- dds[keep,]\n",
    "\n",
    "dds$Target = relevel(dds$Target, ref = \"NT\")\n",
    "\n",
    "dds$Target <- droplevels(dds$Target)\n",
    "\n",
    "dds <- DESeq(dds)\n",
    "\n",
    "write.table(file=\"data/rna/normalized.tsv\", x=as.data.frame(counts(dds, normalized=TRUE)), sep=\"\\t\", quote = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "reading in files with read_tsv\n",
      "\n",
      "1 \n",
      "2 \n",
      "3 \n",
      "4 \n",
      "5 \n",
      "6 \n",
      "7 \n",
      "8 \n",
      "9 \n",
      "\n",
      "\n",
      "summarizing abundance\n",
      "\n",
      "summarizing counts\n",
      "\n",
      "summarizing length\n",
      "\n",
      "using counts and average transcript lengths from tximport\n",
      "\n",
      "estimating size factors\n",
      "\n",
      "using 'avgTxLength' from assays(dds), correcting for library size\n",
      "\n",
      "estimating dispersions\n",
      "\n",
      "gene-wise dispersion estimates\n",
      "\n",
      "mean-dispersion relationship\n",
      "\n",
      "-- note: fitType='parametric', but the dispersion trend was not well captured by the\n",
      "   function: y = a/x + b, and a local regression fit was automatically substituted.\n",
      "   specify fitType='local' or 'mean' to avoid this message next time.\n",
      "\n",
      "final dispersion estimates\n",
      "\n",
      "fitting model and testing\n",
      "\n",
      "2 rows did not converge in beta, labelled in mcols(object)$betaConv. Use larger maxit argument with nbinomWaldTest\n",
      "\n",
      "using 'apeglm' for LFC shrinkage. If used in published research, please cite:\n",
      "    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for\n",
      "    sequence count data: removing the noise and preserving large differences.\n",
      "    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "out of 19927 with nonzero total read count\n",
      "adjusted p-value < 0.1\n",
      "LFC > 0 (up)       : 1141, 5.7%\n",
      "LFC < 0 (down)     : 1151, 5.8%\n",
      "outliers [1]       : 15, 0.075%\n",
      "low counts [2]     : 4243, 21%\n",
      "(mean count < 7)\n",
      "[1] see 'cooksCutoff' argument of ?results\n",
      "[2] see 'independentFiltering' argument of ?results\n",
      "\n"
     ]
    }
   ],
   "source": [
    "\n",
    "idx = (metaData$Target %in% c('NT', 'SMARCC2'))\n",
    "\n",
    "txi = tximport(fns[idx], type = \"salmon\",tx2gene = tx2gene)\n",
    "\n",
    "dds <- DESeqDataSetFromTximport(txi, metaData[idx,], ~Target)\n",
    "\n",
    "keep <- rowSums(counts(dds)) >= 10\n",
    "dds <- dds[keep,]\n",
    "\n",
    "dds$Target = relevel(dds$Target, ref = \"NT\")\n",
    "\n",
    "dds$Target <- droplevels(dds$Target)\n",
    "\n",
    "dds <- DESeq(dds)\n",
    "\n",
    "resLFC = lfcShrink(dds, coef=\"Target_SMARCC2_vs_NT\",type=\"apeglm\")\n",
    "summary(resLFC)\n",
    "\n",
    "\n",
    "write.table(file=\"data/rna/deg/SMARCC2.tsv\", x=as.data.frame(resLFC), sep=\"\\t\", quote = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>TRUE</li><li>TRUE</li><li>TRUE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>FALSE</li><li>TRUE</li><li>TRUE</li><li>TRUE</li><li>TRUE</li><li>TRUE</li><li>TRUE</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item FALSE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\item TRUE\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. FALSE\n",
       "2. FALSE\n",
       "3. FALSE\n",
       "4. FALSE\n",
       "5. FALSE\n",
       "6. FALSE\n",
       "7. FALSE\n",
       "8. FALSE\n",
       "9. FALSE\n",
       "10. FALSE\n",
       "11. FALSE\n",
       "12. FALSE\n",
       "13. FALSE\n",
       "14. TRUE\n",
       "15. TRUE\n",
       "16. TRUE\n",
       "17. FALSE\n",
       "18. FALSE\n",
       "19. FALSE\n",
       "20. FALSE\n",
       "21. FALSE\n",
       "22. FALSE\n",
       "23. TRUE\n",
       "24. TRUE\n",
       "25. TRUE\n",
       "26. TRUE\n",
       "27. TRUE\n",
       "28. TRUE\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE\n",
       "[13] FALSE  TRUE  TRUE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE  TRUE  TRUE\n",
       "[25]  TRUE  TRUE  TRUE  TRUE"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "idx"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.1"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
